inttest/cnichange/cnichange_test.go

```diff
@@ -17,48 +17,79 @@ limitations under the License.
 package cnichange
 
 import (
+	"strings"
 	"testing"
 
 	"github.com/stretchr/testify/suite"
 
 	"github.com/k0sproject/k0s/inttest/common"
+
+	"github.com/k0sproject/k0s/pkg/kubernetes/watch"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/fields"
 )
 
 type CNIChangeSuite struct {
 	common.FootlooseSuite
 }
 
 func (s *CNIChangeSuite) TestK0sGetsUpButRejectsToChangeCNI() {
+	s.PutFile(s.ControllerNode(0), "/tmp/k0s.yaml", k0sConfigWithKubeRouter)
+
 	// Run controller with defaults only --> kube-router in use
-	s.NoError(s.InitController(0))
+	s.NoError(s.InitController(0, "--config=/tmp/k0s.yaml"))
+
+	kc, err := s.KubeClient(s.ControllerNode(0))
+	s.Require().NoError(err)
+
+	// Wait till we see kube-router DS created
+	watch.DaemonSets(kc.AppsV1().DaemonSets("kube-system")).
+		WithObjectName("kube-router").
+		Until(s.Context(), func(ds *appsv1.DaemonSet) (bool, error) {
+			return true, nil
+		})
```

Comment:
Perhaps use `common.WaitForKubeRouterReady` instead?

Reply:
WE cannot in this test case as there's no workers --> it will not get ready ever. :)